### PR TITLE
Upgrade to 2018 edition

### PR DESCRIPTION
**Pull request by jack-fortanix**
This only updates `mbedtls` and not `mbedtls-sys` because the bindings get generated to use `::types::...` when they should use `crate::types::...` in 2018. I couldn't figure out a way to fix this. It probably requires upgrading bindgen.

### CODE REVIEW

File: mbedtls-sys/Cargo.toml [hunk 1]
```diff
@@ -4,6 +4,7 @@ version = "2.18.0"
 authors = ["Jethro Beekman <jethro@fortanix.com>"]
 build = "build/build.rs"
 license = "Apache-2.0/GPL-2.0+"
+#edition = "2018"
```
Comment (Goirad) [inline mbedtls-sys/Cargo.toml]: Can this be uncommented now?

Comment (jack-fortanix) [reply mbedtls-sys/Cargo.toml]: No due to issue with bindgen output mentioned in summary

Comment (Goirad) [reply mbedtls-sys/Cargo.toml]: Oh right sorry, thoguh this was in mbedtls no mbedtls-sys

File: Cargo.lock [hunk 2]
```diff
@@ -206,7 +206,7 @@ dependencies = [
  "cc 1.0.37 (registry+https://github.com/rust-lang/crates.io-index)",
  "core_io 0.1.20190701 (registry+https://github.com/rust-lang/crates.io-index)",
  "libc 0.2.60 (registry+https://github.com/rust-lang/crates.io-index)",
- "mbedtls-sys-auto 2.16.0",
+ "mbedtls-sys-auto 2.18.0",
```
Comment (jethrogb) [inline Cargo.lock]: How come this is only changing now?

Comment (jack-fortanix) [reply Cargo.lock]: No good reason. Was missed in 2.18 upgrade

**Pull request merged**